cmd/sriovdp/sriov-device-plugin.go

```diff
@@ -400,7 +400,6 @@ func (sm *sriovManager) ListAndWatch(empty *pluginapi.Empty, stream pluginapi.De
 	for {
 		select {
 		case <-time.After(10 * time.Second):
-			continue
```

Comment:
Good catch! 
But now we are not doing anything on `case <-time.After(10 * time.Second):`. I think solution to the problem better solved by moving the Probe() before the event handling something like this:

```
	for {
		if sm.Probe() {
			resp := new(pluginapi.ListAndWatchResponse)
			for _, dev := range sm.devices {
				resp.Devices = append(resp.Devices, &pluginapi.Device{ID: dev.ID, Health: dev.Health})
			}
			glog.Infof("ListAndWatch: send devices %v\n", resp)
			if err := stream.Send(resp); err != nil {
				glog.Errorf("Error. Cannot update device states: %v\n", err)
				sm.grpcServer.Stop()
				return err
			}
		}
		
		select {
		case <-time.After(10 * time.Second):
			continue
		case <-sm.termSignal:
			glog.Infof("Terminate signal received, exiting ListAndWatch.")
			return nil
		}
	}
```
This way we can ensure Probe() invoked first when we get into to event watch loop and 10 seconds after we Probe() again. Let me know what you think

Reply:
> Good catch!
> But now we are not doing anything on `case <-time.After(10 * time.Second):`. I think solution to the problem better solved by moving the Probe() before the event handling something like this:
> 
> ```
> 	for {
> 		if sm.Probe() {
> 			resp := new(pluginapi.ListAndWatchResponse)
> 			for _, dev := range sm.devices {
> 				resp.Devices = append(resp.Devices, &pluginapi.Device{ID: dev.ID, Health: dev.Health})
> 			}
> 			glog.Infof("ListAndWatch: send devices %v\n", resp)
> 			if err := stream.Send(resp); err != nil {
> 				glog.Errorf("Error. Cannot update device states: %v\n", err)
> 				sm.grpcServer.Stop()
> 				return err
> 			}
> 		}
> 		
> 		select {
> 		case <-time.After(10 * time.Second):
> 			continue
> 		case <-sm.termSignal:
> 			glog.Infof("Terminate signal received, exiting ListAndWatch.")
> 			return nil
> 		}
> 	}
> ```
> This way we can ensure Probe() invoked first when we get into to event watch loop and 10 seconds after we Probe() again. Let me know what you think

Move Probe() on top is an option, the reason why I didn't do that is because when ListAndWatch is invoked, an initial ListAndWatchResponse is sent back immediately to kubelet, there seems no need to do the Probe() right after the initial response;  but I admit that this is not a big overhead and will only happen once.